external/model-preparation-algorithm/mpa_tasks/apis/task.py

```diff
@@ -181,6 +184,34 @@ def _initialize(self, dataset=None, output_model=None, export=False):
                 del self._model_cfg['fp16']
         self._precision = [ModelPrecision.FP32]
 
+        # Add/remove adaptive interval hook
+        if 'use_adaptive_interval' in self._recipe_cfg:
+            if self._recipe_cfg.use_adaptive_interval is True:
```

Comment:
I don't think we need this if statement.

Reply:
What statement are you talking? You reviewed the same part as below.

Reply:
I can't remove that if statement. Otherwise, it will cause error in the export stage. initialize function where init_recipe_hparam is initiated is called only when it's not export.

Reply:
What I mean is that if statement right above and this if statement can be combined into one, and I don't think it's going to cause an error because of this during export, so what do you think?

Reply:
Your suggestion is really good, but it raises error in else for cls/seg tasks -> self._recipe_cfg.pop('adaptive_validation_interval')

use_adaptive_interval is the parameter only for detection.(because it did not work in other tasks.. must test at other time) So it needs to check first if use_adaptive_interval is actually in the recipe_cfg. and then need to pop and remove adaptive_validation_interval config only for det task when use_adaptive_interval is set to False.

Reply:
I think we can prevent errors by using `self._recipe_cfg.pop('adaptive_validation_interval', None)`.

Reply:
I modified this part. Please check it again!